### PR TITLE
Addressing typos and elaboration requests in the spec.

### PR DESCRIPTION
**Pull request by viktorklang**
See #407

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -80,6 +80,7 @@ followed by a possibly unbounded number of `onNext` signals (as requested by `Su
 | <a name="term_terminal_state">Terminal state</a> | For a Publisher: When `onComplete` or `onError` has been signalled. For a Subscriber: When an `onComplete` or `onError` has been received.|
 | <a name="term_nop">NOP</a> | Execution that has no detectable effect to the calling thread, and can as such safely be called any number of times.|
 | <a name="term_ext_sync">External synchronization</a> | Access coordination for thread safety purposes implemented outside of the constructs defined in this specification, using techniques such as, but not limited to, `atomics`, `monitors`, or `locks`. |
+| <a name="term_thread-safe">Thread-safe</a> | Can be safely invoked synchronously, or asychronously, without requiring external synchronization to ensure program correctness. |
```
Comment (viktorklang) [inline README.md]: @reactive-streams/contributors I'm sure this definition can be improved upon, so please suggest improvements. :)

**Comment by pavelrappo** [comment]
Looks good, thanks!

**Comment by viktorklang** [comment]
@reactive-streams/contributors Yay or nay?

**Comment by DougLea** [comment]
Looks OK to me. I plan to also include Pavel's suggestion about adding a happens-before clause to j.u.c.SubmissionPublisher, since it holds for that class in particular.

File: README.md [hunk 2]
```diff
@@ -97,7 +98,7 @@ public interface Publisher<T> {
...
 | <a name="1.2">2</a>       | A `Publisher` MAY signal fewer `onNext` than requested and terminate the `Subscription` by calling `onComplete` or `onError`. |
 | [:bulb:](#1.2 "1.2 explained") | *The intent of this rule is to make it clear that a Publisher cannot guarantee that it will be able to produce the number of elements requested; it simply might not be able to produce them all; it may be in a failed state; it may be empty or otherwise already completed.* |
-| <a name="1.3">3</a>       | `onSubscribe`, `onNext`, `onError` and `onComplete` signaled to a `Subscriber` MUST be signaled in a `thread-safe` manner—and if performed by multiple threads—use [external synchronization](#term_ext_sync). |
+| <a name="1.3">3</a>       | `onSubscribe`, `onNext`, `onError` and `onComplete` signaled to a `Subscriber` MUST be signaled in a [thread-safe](#term_thread-safe) manner—and if performed by multiple threads—use [external synchronization](#term_ext_sync). |
```
Comment (viktorklang) [inline README.md]: @reactive-streams/contributors Will the definition of *thread-safe* be in conflict with this clause about *external synchronization* or is it clear what it means here?

File: README.md (hunk 1 above)
Comment (viktorklang) [reply README.md]: Is it clear that this doesn't imply that it is safe to be called *in parallel*?

**Pull request merged**